client/jetpack-connect/signup.js

```diff
@@ -202,7 +199,11 @@ export class JetpackSignup extends Component {
 	handleUserCreationError = ( error ) => {
 		const { errorNotice, translate, warningNotice } = this.props;
 		debug( 'Signup error: %o', error );
-		this.resetState();
+		this.setState( {
+			newUsername: null,
+			bearerToken: null,
+			isCreatingAccount: false,
+		} );
```

Comment:
As I understand those lines reset everything except `wooDnaFormType`, right? If so I'd suggest using lodash omit for clarity.

Reply:
The state of this component got a bit more complicate with this flow. [These](https://github.com/Automattic/wp-calypso/blob/515e63b869fb4bb13a4b3f0e46764254d57132dd/client/jetpack-connect/signup.js#L119-L124) are all the properties that shouldn't be reset.

Reply:
If we have something like this, will it be better? In other case it's not so clear why those changes need to be applied to the state.
```js
this.setState( omit( this.constructor.initialState, [ 'wooDnaFormType' ] ) );
```

Anyway, let's not spend too much time on this. Any option will work for now and can be improved later if required.

Reply:
I'd rather remove the `initialState` variable alltogether, since it no longer makes sense to use: https://github.com/Automattic/wp-calypso/pull/43270/commits/bfad4e5aa56c9b4b94b4d6da259f92e016bfe8ea

> In other case it's not so clear why those changes need to be applied to the state.

Ideally, the state update should make sense on its own. It's either a "full component state reset" (that's how it was before), or it's not, and in that case the properties that need updating should be explicit. I understand why `isCreatingAccount` must be set to `false`, and why `newUsername` needs to be reset too, but honestly `bearerToken` shouldn't need to be reset. But I don't know enough about this code to touch it more than the bare minimum :)